src/non_blocking/deflate/decode.rs

```diff
@@ -129,7 +129,7 @@ impl<R: Read> Read for Decoder<R> {
                     let buf = &mut buf[..cmp::min(buf_len, *len as usize)];
                     read_size = self.bit_reader.as_inner_mut().read(buf)?;
 
-                    self.block_decoder.buffer.extend(&buf[..read_size]);
+                    self.block_decoder.extend(&buf[..read_size]);
```

Comment:
It looks like we can entirely remove this line?

Reply:
I think the line is required because if the next block succeeding a non-compressed block is compressed, there is a possibility that the compressed block may refers some strings in the previous non-compressed block (the referents must be held in `block_decoder.buffer`).